provider/core/src/data_provider.rs

```diff
@@ -206,38 +242,32 @@ where
     /// # } // feature = "serde_json"
     /// ```
     #[allow(clippy::type_complexity)]
-    pub fn try_from_rc_buffer_badly<E>(
-        rc_buffer: Rc<[u8]>,
+    pub fn try_from_buffer_badly<E>(
```

Comment:
Nit: You currently are inconsistent: you have `try_from_rc_buffer` and `try_from_buffer_badly`.

I have a preference for keeping the `_rc` in the function name.